src/test/suite/lib/run-ggshield.test.ts

```diff
@@ -0,0 +1,30 @@
+import * as simple from "simple-mock";
+import * as childProcess from "child_process";
+import * as runGGShield from "../../../lib/run-ggshield";
+import assert = require("assert");
+import { GGShieldConfiguration } from "../../../lib/ggshield-configuration";
+
+suite("runGGShieldCommand", () => {
+  let spawnSyncMock: simple.Stub<Function>;
+
+  setup(() => {
+    spawnSyncMock = simple.mock(childProcess, "spawnSync"); // Mock spawnSync
+  });
+
+  teardown(() => {
+    simple.restore();
+  });
+
+  test("Global env variables are set correctly", async () => {
+    process.env.TEST_GLOBAL_VAR = "GlobalValue";
+
+    spawnSyncMock.callFn((command, args, options) => {
+      assert.strictEqual(options.env.TEST_GLOBAL_VAR, "GlobalValue");
+      return {};
+    });
```

Comment:
I'm not sure I like this since it means the test will succeed if spawnSync is never called.
Could we instead juste assert that we passed an "env" with the right env vars in the options of spawnSync ?